src/coreclr/vm/frozenobjectheap.cpp

```diff
@@ -33,6 +36,7 @@ Object* FrozenObjectHeapManager::TryAllocateObject(PTR_MethodTable type, size_t
 #else // FEATURE_BASICFREEZE
 
     Object* obj = nullptr;
+    FrozenObjectSegment* currentSegment = nullptr;
     {
         CrstHolder ch(&m_Crst);
```

Comment:
```suggestion
        GCX_PREEMP();

        CrstHolder ch(&m_Crst);
```
Would it be possible to switch to preemptive mode around this lock? This is potentially expensive operation due to VirtualAlloc calls. It is better to be in preemptive mode for that; and to take unmanaged locks in general - preemptive mode makes sure that the lock won't cause GC starvation or deadlock with GC.